code/_onclick/item_attack.dm

```diff
@@ -40,17 +44,17 @@
 
 	var/attackby_result
 	if (is_right_clicking)
-		switch (target.attackby_secondary(src, user, params))
+		switch (target.attackby_secondary(source_atom, user, params))
 			if (SECONDARY_ATTACK_CALL_NORMAL)
-				attackby_result = target.attackby(src, user, params)
+				attackby_result = target.attackby(source_atom, user, params)
```

Comment:
If attackby abides by the proxy, pre-attack should as well.

Reply:
Can't because preattack is called on `src` & not on `target`. `source_atom` is the proxy that will act on `target` on behalf of us(i.e. `src`). If we did `preattack(source_atom)` then we would be checking "are we attacking our own proxy?" which is non sensical